### PR TITLE
Network SDK with API version 2018-05-01

### PR DESCRIPTION
**Pull request by anton-evseev**
Based on: https://github.com/Azure/azure-rest-api-specs/pull/3089

## Description

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by MikhailTryakhov** [comment]
hi @dsgouda please review new major SDK version. Address all the comments for me so I can help with the fixes

File: src/SDKs/_metadata/network_resource-manager.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
-2018-04-26 20:59:33 UTC
+2018-05-17 13:34:00 UTC
```
Comment (dsgouda) [inline src/SDKs/_metadata/network_resource-manager.txt]: @MikhailTryakhov please generate the sdk using [this](https://github.com/ritwikbasu/azure-sdk-for-net/blob/21e04d84695b939a000037b76757addf664ba8c4/src/SDKs/Network/Management.Network/generate.ps1) script instead

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks great apart from the minor comment.

**Review by dsgouda — APPROVED** [review]
Approving this for now since some teams need the nuget package soon

**Pull request merged**